analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RoslynSymbolicExecution.cs

```diff
@@ -92,45 +92,44 @@ private IEnumerable<ExplodedNode> ProcessBranching(ExplodedNode node)
             {
                 checks.ExitReached(new(null, node.State));
             }
-            else if (node.Block.ContainsThrow())
-            {
-                yield return new(cfg.ExitBlock, node.State, null);
-            }
             else
             {
                 foreach (var successor in node.Block.Successors.Where(x => IsReachable(node, x)))
                 {
-                    if (ProcessBranch(node, successor) is { } newNode)
+                    foreach (var successorNode in ProcessBranch(node, successor).WhereNotNull())
                     {
-                        yield return newNode;
+                        yield return successorNode;
                     }
                 }
             }
         }
 
-        private ExplodedNode ProcessBranch(ExplodedNode node, ControlFlowBranch branch)
+        private IEnumerable<ExplodedNode> ProcessBranch(ExplodedNode node, ControlFlowBranch branch)
```

Comment:
Ah, this change is unfortunate. I didn't foresee it. It makes cardinality much more complicated when processing a single branch produces a collection. 
It also removes the explicit `// We don't know where to continue` scenario - this will be an important hint for us in the future to feel safe about extending the `else if` tree.

Can we move the code back to `ProcessBranching`? I'm not very happy about it, but it will make the overall structure better. To have the complex branching in `*Branching` and single branch in `*Branch`